backend/src/controllers/AuthController.js

```diff
@@ -1,30 +1,84 @@
+import ResetTokenRepo from "../database/repositories/ResetTokenRepo.js";
+import UserRepo from "../database/repositories/UserRepo.js";
+import {NotFoundError} from "../lib/errors/index.js";
+import { sendResetLink } from "../services/user/password_reset.js";
+import bcrypt from "bcrypt";
+
 export default class AuthController {
-  static loginStatus = async (req, res) => {
-    if (req.user) {
-      res.status(200).json({
-        success: true,
-        message: 'Login was successful',
-        user: req.user,
-      });
-    } else {
-      res.status(401).json({
-        success: false,
-        message: 'Not Authenticated',
-      });
-    }
-  };
-  static loginFailed = async (req, res) => {
-    res.status(401).json({
-      success: false,
-      message: 'Not Authenticated',
-    });
-  };
-  static logout = async (req, res) => {
-    req.logout(function (err) {
-      if (err) {
-        return next(err);
-      }
-      return res.status(200).json({ success: true, message: 'Logout Successful' });
-    });
-  };
-}
+    static loginStatus = async (req, res) => {
+        if (req.user) {
+            res.status(200).json({
+                success: true,
+                message: "Login was successful",
+                user: req.user,
+            });
+        } else {
+            res.status(401).json({
+                success: false,
+                message: "Not Authenticated",
+            });
+        }
+    };
+    static loginFailed = async (req, res) => {
+        res.status(401).json({
+            success: false,
+            message: "Not Authenticated",
+        });
+    };
+    static logout = async (req, res) => {
+        req.logout();
+        res.status(200).json({
+            success: true,
+            message: "Logout successful",
+        });
+    };
+
+    static getResetLink = async (req, res) => {
+        const {email} = req.body;
+
+        try {
+            const registeredUser = await UserRepo.getUserByEmail(email);
+
+            if (!registeredUser) throw new NotFoundError("Please input a valid registered email.");
+
+            const {name, id} = registeredUser;
+
+            sendResetLink(email, name, id);
+
+            res.status(200).redirect("/login");
+            return ({message: "A password reset link has been sent to your email address"});
+        
+        } catch (error) {
+            console.error(error);
+        }
+    };
+
+    static updateUserPassword = async (req, res) => {
+        const {token, id, password} = req.body;
```

Comment:
You sent these out as request params, now you're expecting it from request body.

Reply:
No. Not really. The frontend guy retrieves them from the url and puts them into the payload alongside the new password to be set. I've already explained that to @dread557